strapi/strapi
# feat(content-releases): add scheduledAt to releases content type & requests

⚠️ This PR is supposed to aiming develop, but is going to point to `feat/scheduling-service` until #19414 is merged. Please take a look there if you haven't already. 😄 

### What does it do?

* Adds a new attribute to the releases schema: `scheduledAt`, used to save the publish date for a release.
* Now the create release endpoint accepts the scheduledAt as well; this attribute is optional.
* If scheduledAt is provided when creating a release, we check that the scheduledDate is later than the current moment with a new validator on the validation service.
* If scheduledAt is provided and is valid, we schedule the cron job to plan a release.

## Testing

### Enabling the future flags

The scheduling functionality is implemented through future flags. To begin, ensure the future flag is set to true. Navigate to `config/features.js` and include `contentReleasesScheduling` as an option within your future object. Create the future object if it does not exist.

Example:
```js
// config/features.js

module.exports = ({ env }) => ({
  future: {
    contentReleasesScheduling: true,
  },
});

```

### How to test it?

Valid path:

1. You should create a new release with a scheduled datetime, and the release should be published at that datetime. ⚠️ Please be aware that all times in the backend are handled in the UTC timezone, so consider that when sending a datetime directly to the endpoint.

Error handling:
1. If you try to create a release with an invalid date in the scheduledAt field, it should throw an error.
